Edits to minimum bar (#1)
CTF-Organizers/Wiki: minimum_bar.md

In this page you can find a list of minimum requirements that every CTF competition should have. The target audience for this document is CTF organizers that wish to organize a CTF competition that follows the expectations of players in the community. If a requirement can't be met for some reason, players should know.

# Topics Covered:

> Each of the topics covered include a summary of what the topic is as well as a set of minimum requirements to ensure a quality CTF.

Reviewer: ```suggestion
Each of the topics covered include a summary of what the topic is about, as well as a set of minimum requirements to ensure a minimum quality CTF.
```

1. Format and Mechanics
2. Task Quality
3. Fairness

# Format and Mechanics
CTF competitions follow well understood mechanics, as such, it is important that when someone announces a CTF competition open to the public it satisfies players expectations. In specific, when someone uses the terms **Jeopardy-style CTF** or **Attack-Defense CTF**, they should follow the mechanics expected in such competitions. Deviations from these mechanics should be explicitly declared up-front, so players are not surprised when they start playing.

The minimum bar **requirements** for **all CTF competitions** are: 

> * The competition should be announced to players early on (at least two weeks ahead is suggested)

Reviewer: ```suggestion
* The competition should be announced to players early on (at least two weeks ahead is suggested).
```

* Include the categories of the competition, an approximate number of challenges, and a definite starting and ending time 

> * Organizers should provide a channel in which players are able to quickly reach out to and communicate problems (with tasks or infrastructure). Examples of this include a Discord Guild or a Slack Workspace.

Reviewer: ```suggestion
* Organizers should provide a channel in which players are able to quickly reach out to and communicate problems (with tasks or infrastructure). Examples of this include an IRC channel, a Discord Guild or a Slack Workspace.
```

* Usually the list of teams that solved each task is public. In those cases, players usually expect to be able to know how many teams are actively participating. As such, having a task that players just have to copy-paste to mark they are active is helpful.
* The duration of CTF competitions usually is 48 hours during a weekend. There are exceptions to this rule, but that will change the expectations of the teams playing.
* Most CTFs have challenges in categories Binary Exploitation (pwn), Web Exploitation (web), Cryptography (crypto), and Reverse Engineering (re). It is possible for a task to be in more than one category.

> * Tasks should be scored using static scoring (organizers define number of points ahead of time), or a known and tested dynamic scoring formula (points changes based on number of solves). At this moment, there are pros and cons to both. If you are unsure about the weights of challenges, it is reccomended to use a dynamic scoring formula.

Reviewer: ```suggestion
* Tasks should be scored using static scoring (organizers define number of points ahead of time), or a known and tested dynamic scoring formula (points changes based on number of solves). At this moment, there are pros and cons to both. If you are unsure about the weights of challenges, it is recommended to use a dynamic scoring formula.
```

> * Tasks should be scored using static scoring (organizers define number of points ahead of time), or a known and tested dynamic scoring formula (points changes based on number of solves). At this moment, there are pros and cons to both. If you are unsure about the weights of challenges, it is reccomended to use a dynamic scoring formula.

Reviewer: we probably should then link to a dynamic scoring formula?

* All information should be public to all players at the same time.
* Any communication from organizer to player that can hint to any new information should, without exception, also be known to all players. 
* If the competition is online and global, organizers should make sure that regardless of the timezone a player uses, they should have the same ability to earn points (in the case first-blood bonus points or speed-runs are used).

# Stability
Given the time-limited aspect of CTF competitions, a slight disruption on the availability of a task can affect the fairness and experience of the competition. Either malicious (DoS) or accidental downtime should be planned for ahead of time by the organizers, and some minimum requirements regarding the stability of the infrastructure, including both the scoreboard and the tasks.

The minimum **requirements** for **stability** are: 
* Organizers should have an independent communication channel to players (IRC, or similar) independent of their own infrastructure. 
* Links to the tasks that don't require online services for working on (eg, reversing, pwnable) should be published in case of long-term downtime.

>     * It is **highly reccomended** to use some sort of object storage service such as AWS S3 or Google Drive to serve these independently of your servers.

Reviewer: ```suggestion
    * It is **highly recommended** to use some sort of object storage service such as AWS S3, Google Cloud Storage, Dropbox, to serve these independently of the competition infrastructure.
```
